### PR TITLE
Add gke data to inventory

### PR DESCRIPTION
**Pull request by blueandgold**
A couple of noteworthy comments:

- What should we do with master_auth data?  It has sensitive password/cert info.  I am defaulting to saving this into mysql.  But will be happy to hear of any suggestion to handle this better.

- Saving server_config into the same table as the rest of the gke cluster data.  Seems like a 1:1 mapping, so we can save a table here.

### CODE REVIEW

File: tests/inventory/pipelines/test_data/fake_appengine_applications.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Fake AppEngine application data."""
+"""Fake GKE service data."""
```
Comment (ahoying) [inline tests/inventory/pipelines/test_data/fake_appengine_applications.py]: Update the wrong file?

File: google/cloud/security/common/data_access/sql_queries/create_tables.py [hunk 2]
```diff
@@ -313,6 +313,40 @@
...
+        `legacy_abac` json DEFAULT NULL,
+        `locations` varchar(255) DEFAULT NULL,
+        `logging_service` varchar(255) DEFAULT NULL,
+        `master_auth` json DEFAULT NULL,
+        `monitoring_service` varchar(255) DEFAULT NULL,
+        `name` varchar(255) DEFAULT NULL,
+        `network` varchar(255) DEFAULT NULL,
+        `node_config` json DEFAULT NULL,
+        `node_ipv4_cidr_size` int DEFAULT NULL,
+        `node_pools` varchar(4096) DEFAULT NULL,
```
Comment (ahoying) [inline google/cloud/security/common/data_access/sql_queries/create_tables.py]: node_pools is a repeated field that can be much larger than 4k. This should be a json field, not a varchar.

File: google/cloud/security/inventory/pipelines/load_gke_pipeline.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+                       'logging_service': cluster.get('loggingService'),
+                       'master_auth': parser.json_stringify(
+                            cluster.get('masterAuth')),
+                       'monitoring_service': cluster.get('monitoringService'),
+                       'name': cluster.get('name'),
+                       'network': cluster.get('network'),
+                       'node_config': parser.json_stringify(
+                            cluster.get('nodeConfig')),
+                       'node_ipv4_cidr_size': cluster.get('nodeIpv4CidrSize'),
+                       'node_pools': cluster.get('nodePools'),
```
Comment (ahoying) [inline google/cloud/security/inventory/pipelines/load_gke_pipeline.py]: probably need to json.stringify this too.

File: google/cloud/security/inventory/api_map.py [hunk 4]
```diff
@@ -57,6 +57,12 @@
             'class_name': 'StorageClient',
             'version': None
         },
+    'gke_api':
```
Comment (ahoying) [inline google/cloud/security/inventory/api_map.py]: Doesn't 'container' need to be added to configs/forseti_conf.yaml.* too?

Comment (blueandgold) [reply google/cloud/security/inventory/api_map.py]: Good catch, fixed in the conf yaml files.

File: google/cloud/security/common/data_access/sql_queries/create_tables.py (hunk 2 above)
Comment (blueandgold) [reply google/cloud/security/common/data_access/sql_queries/create_tables.py]: Done.

File: google/cloud/security/inventory/pipelines/load_gke_pipeline.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+        gke_services = {}
+        for project in projects:
+            clusters = self.safe_api_call('get_clusters', project.id)
+
+            if clusters:
+            
+                for cluster in clusters:
+                    self_link_parts = cluster.get('selfLink').split('/')
+                    zone = self_link_parts[self_link_parts.index('zones')+1]
+                    server_config = self.safe_api_call(
```
[the remark this answers is not in the document]
Comment (blueandgold) [reply google/cloud/security/inventory/pipelines/load_gke_pipeline.py]: Putting a TODO.  I think that in practice, folks should run in different zones for redundancy.

File: google/cloud/security/inventory/pipelines/load_gke_pipeline.py [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                       'current_node_count': cluster.get('currentNodeCount'),
+                       'current_node_version': cluster.get('currentNodeVersion'),
+                       'endpoint': cluster.get('endpoint'),
+                       'initial_cluster_version': cluster.get('initialClusterVersion'),
+                       'instance_group_urls': cluster.get('instanceGroupUrls'),
+                       'legacy_abac': parser.json_stringify(
+                            cluster.get('legacyAbac')),
+                       'locations': cluster.get('locations'),
+                       'logging_service': cluster.get('loggingService'),
+                       'master_auth': parser.json_stringify(
```
[the remark this answers is not in the document]
Comment (blueandgold) [reply google/cloud/security/inventory/pipelines/load_gke_pipeline.py]: Done, not saving this.  Setting it with a message that forseti will not save it.

File: google/cloud/security/inventory/pipelines/load_gke_pipeline.py (hunk 3 above)
Comment (blueandgold) [reply google/cloud/security/inventory/pipelines/load_gke_pipeline.py]: Done.

File: tests/inventory/pipelines/test_data/fake_gke_services.py [hunk 7]
```diff
@@ -153,32 +153,14 @@
...
-                     'selfLink': 'https://container.googleapis.com/v1/projects/cicd-prod/zones/us-central1-b/clusters/redditmobile-canary-central/nodePools/default-pool',
-                     'status': 'RUNNING',
-                     'version': '1.4.5'}],
+     'node_pools': '[{"status": "RUNNING", "version": "1.4.5", "name": "default-pool", "instanceGroupUrls": ["https://www.googleapis.com/compute/v1/projects/cicd-prod/zones/us-central1-b/instanceGroupManagers/gke-redditmobile-canary--default-pool-1388e059-grp"], "management": {}, "autoscaling": {}, "config": {"diskSizeGb": 100, "machineType": "n1-standard-1", "oauthScopes": ["https://www.googleapis.com/auth/compute", "https://www.googleapis.com/auth/devstorage.read_only", "https://www.googleapis.com/auth/logging.write", "https://www.googleapis.com/auth/servicecontrol", "https://www.googleapis.com/auth/service.management.readonly", "https://www.googleapis.com/auth/trace.append"], "imageType": "COS", "serviceAccount": "default"}, "selfLink": "https://container.googleapis.com/v1/projects/cicd-prod/zones/us-central1-b/clusters/redditmobile-canary-central/nodePools/default-pool", "initialNodeCount": 1}]',
```
Comment (ahoying) [inline tests/inventory/pipelines/test_data/fake_gke_services.py]: The names of the project and gke cluster and node pools in the test should be better anonymized.

File: tests/inventory/pipelines/test_data/fake_gke_services.py [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+       'networkPolicyConfig':{  
+          'disabled': True
+       }
+    },
+    'clusterIpv4Cidr': '10.60.0.0/14',
+    'createTime': '2016-11-12T17:57:49+00:00',
+    'currentMasterVersion': '1.6.13-gke.1',
+    'currentNodeCount':2,
+    'currentNodeVersion': '1.4.5',
+    'endpoint': '104.197.231.35',
```
Comment (ahoying) [inline tests/inventory/pipelines/test_data/fake_gke_services.py]: This should be a fake IP

File: tests/inventory/pipelines/test_data/fake_gke_services.py [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+    'initialClusterVersion': '1.4.5',
+    'instanceGroupUrls':[  
+       'https://www.googleapis.com/compute/v1/projects/cicd-prod/zones/us-central1-b/instanceGroupManagers/gke-redditmobile-canary--default-pool-1388e059-grp'
```
Comment (ahoying) [inline tests/inventory/pipelines/test_data/fake_gke_services.py]: The project name, cluster name and other details should be fake / anonymous.

Comment (blueandgold) [reply tests/inventory/pipelines/test_data/fake_gke_services.py]: Done.  Good spot.

File: tests/inventory/pipelines/test_data/fake_gke_services.py (hunk 8 above)
Comment (blueandgold) [reply tests/inventory/pipelines/test_data/fake_gke_services.py]: Done.

**Pull request merged**